### DISCUSSION TITLE
xDrip as Nightscout follower may receive readings with an extra 5-minute delay

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#3164] [Bug reports]
After seeing this open PR (https://github.com/NightscoutFoundation/xDrip/pull/1307) and the issue it attempts to address (https://github.com/NightscoutFoundation/xDrip/issues/1302), and this open issue (https://github.com/NightscoutFoundation/xDrip/issues/929), I believe I have been able to recreate the problem.

I have my active xDrip getting readings from a G6.  It uploads to my main Nightscout.
I set up a test phone to act as a Nightscout follower getting readings from my main Nightscout site.
I set up this same xDrip to upload its readings to a second test Nightscout site.
I set up a third phone to act as a Nightscout follower getting its readings from the second Nightscout site.

The second Nightscout follower shows the delay described in the issues and PR.
![Screenshot_20231102-152053](https://github.com/NightscoutFoundation/xDrip/assets/51497406/bf328939-8f44-4be0-bb8c-abbf2206b621)

**Comment by Navid200** [comment]
I will open a PR to address this.

**Reply by ChiSoxDev** [reply]
Thank you!

**Reply by Navid200** [reply]
@ChiSoxDev 
On my test device, I can see that there are no missing points.  However, the problem is that the last point always arrives 5 minutes too late.
Can you confirm that this is your experience?  Meaning you have no missing points.  In other words, you are not receiving only every other point.  You are receiving all the points.  Except the last reading is 5 minutes too late.  Is that right?

**Comment by Navid200** [comment]
This is now fixed with the latest Nightly release: https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.11.21
It has a new setting when Nighscout follower is selected as the data source.  The new setting lets you delay when Nightscout follower reads from Nightscout.
